coffeecard/CoffeeCard.Library/Services/v2/ProductService.cs

```diff
@@ -57,6 +58,111 @@ public async Task<Product> GetProductAsync(int productId)
+                UserGroup = userGroup
+            }).ToList();
+
+            _context.ProductUserGroups.AddRange(productUserGroups);
+            
+            _context.Products.Add(product);
+            
+            _context.SaveChanges();
+
+
+            var result = new InitiateProductResponse
+            {
+                Price = product.Price,
+                Description = product.Description,
+                Name = product.Name,
+                NumberOfTickets = product.NumberOfTickets,
+                Visible = product.Visible
+            };
+
+            return Task.FromResult(result);
+        }
+        
+        private Product GetProduct(int productId)
+        {
+            return _context.Products.Where(p => p.Id == productId).FirstOrDefault();
+        }
+        
+        public Task<InitiateProductResponse> UpdateProduct(InitiateProductRequest changedProduct)
+        {
+            var product = GetProduct(changedProduct.Id);
+
+            if (changedProduct.Id != default(int))
+            {
+                Log.Information($"Changing Id of product from {product.Id} to {changedProduct.Id}");
+                product.Id = changedProduct.Id;
+            }
+
+            if (changedProduct.NumberOfTickets != default(int))
+            {
+                Log.Information($"Changing NumberOfTickets of product from {product.NumberOfTickets} to {changedProduct.NumberOfTickets}");
+                product.NumberOfTickets = changedProduct.NumberOfTickets;
+            }
+
+            if (!string.IsNullOrEmpty(changedProduct.Name))
+            {
+                Log.Information($"Changing Name of product from {product.Name} to {changedProduct.Name}");
+                product.Name = changedProduct.Name;
+            }
+
+            if (changedProduct.ExperienceWorth != default(int))
+            {
+                Log.Information($"Changing ExperienceWorth of product from {product.ExperienceWorth} to {changedProduct.ExperienceWorth}");
+                product.ExperienceWorth = changedProduct.ExperienceWorth;
+            }
+
+            if (changedProduct.Visible != default(bool))
+            {
+                Log.Information($"Changing Visible of product from {product.Visible} to {changedProduct.Visible}");
+                product.Visible = changedProduct.Visible;
+            }
+
+            _context.SaveChanges();
+            
+            var result = new InitiateProductResponse
+            {
+                Price = product.Price,
+                Description = product.Description,
+                Name = product.Name,
+                NumberOfTickets = product.NumberOfTickets,
+                Visible = product.Visible
+            };
+            
+            return Task.FromResult(result);
+        }
+
+        public async Task<bool> DeactivateProduct(int productId)
+        {
+            var product = await _context.Products.FindAsync(productId);
+
+            if (product == null)
```

Comment:
Typically we throw the EntityNotFoundExceptin when we are unable to find an item, use that instead